### DISCUSSION TITLE
Unfolding the q grid in the prefix.result file

### DISCUSSION THREAD

**Discussion by whzhangg** [ttadano/alamode#15] [Q&A] [answered]
Hi,

I want to interface alamode result with code that does not treat the irreducible q grid, So I want to unfold the irreducible q point to the full BZ using the information from the SYMM_INFO_PRIM

Is this the correct way to perform this operation?
  take the rotation part of sym operation in SYMM_INFO_PRIM by rot(1,1), rot(1,2), .... rot(3,2), rot(3,3)
  find the set of full q grid by inv(rot(: ,:)) dot irr_xq(:) of all symmetry operation

I see in kpoint.cpp that transpose(inv(rot)) is used for the symmetry operation. Is the transpose necessary?

Thank you,
Wenhao

**Comment by ttadano** [comment, marked as answer]
The transpose is necessary.

The rotation matrix _R_  in ``SYMM_INFO_PRIM`` is defined in the lattice coordinate and transforms the input fractional coordinate **_r_** into another equivalent structure **_r'_** = _R_**_r_**. To transform the momentum defined in the reciprocal lattice basis into a symmetrically reducible one, we need to multiply _S_ = (_R_<sup>-1</sup>)<sup>T</sup> as **_k'_** = _S_**_k_**.

When the rotation is performed in the Cartesian coordinate or when the lattice vector is orthogonal, the rotation matrix _R_ becomes orthogonal. In these cases, we have _S_ = _R_.

**Reply by whzhangg** [reply]
I see, if I understand correctly we have **(k2).T * R * r1 =(k1).T * r1** giving **k2 = (R.T)^-1 k1**. 

Also, do you think it's convenient to rotate the group velocity in the result file? Is the group velocity of equivalent q point under #GAMMA_EACH tag also generated by rotation in the alamode code?

**Reply by ttadano** [reply]
The group velocities of symmetrically equivalent **_q_** point shown in the ``PREFIX.result`` file are generated by taking the finite-difference of the phonon frequency at each **_q_** point. This computation is quite cheap and easier to implement than rotating the group velocity. Rotating the group velocity by using the matrix _S_ is, of course, possible and not very difficult.
